locals.tf

```diff
@@ -0,0 +1,35 @@
+# ------------------------------------------------------------------------------
+# Retrieve the effective Account ID, User ID, and ARN in which Terraform is
+# authorized.  This is used to determine the CyHy account ID.
+# ------------------------------------------------------------------------------
+data "aws_caller_identity" "cyhy" {}
+
+# Retrieve the information for all accounts in the organization.  This is used,
+# for instance, to lookup the account ID for the Users account.
+data "aws_organizations_organization" "cool" {
+  provider = aws.organizationsreadonly
+}
+
+# ------------------------------------------------------------------------------
+# Evaluate expressions for use throughout this configuration.
+# ------------------------------------------------------------------------------
+locals {
+  # Get the CyHy account ID.
+  cyhy_account_id = data.aws_caller_identity.cyhy.id
+
+  # Determine if this is a Production workspace by checking if
+  # terraform.workspace begins with "prod"
+  production_workspace = length(regexall("^prod", terraform.workspace)) == 1
+
+  # In production Terraform workspaces, the string '-production' is appended to
+  # the bucket name.  In non-production workspaces, '-<workspace_name>' is
+  # appended to the bucket name.
+  lambda_bucket_name = format("%s-%s", var.cyhy_lambda_artifacts_s3_bucket, local.production_workspace ? "production" : terraform.workspace)
```

Comment:
Since bucket names are global across AWS, I think we still need to do this, otherwise there will be bucket name conflicts.

Additionally, I think we still want to enable multiple developers to spin up their own dev environments within a single CyHy account (probably still using workspaces to separate things, but maybe eventually something else more Terragrunt-y).

Reply:
If we're worried about global names should we consider using the [`bucket_prefix' argument](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/s3_bucket#bucket_prefix)? We can just set the bucket name as an output (which wouldn't be a bad idea even as it is now).

Reply:
Please see https://github.com/cisagov/cool-accounts-cyhy/pull/1/commits/1abf8f029fe06c1a8dbea7e9ec05530d50022572 and https://github.com/cisagov/cool-accounts-cyhy/pull/1/commits/8dd7eda6baceebd60ac8d3fa00c87943141612b5 which should scratch this itch.